src/coreclr/src/System.Private.CoreLib/src/System/Text/StringBuilder.CoreCLR.cs

```diff
@@ -145,7 +145,7 @@ internal unsafe void InternalCopy(IntPtr dest, int len)
         /// <param name="byteIndex">The logical byte index in this builder.</param>
         private StringBuilder FindChunkForByte(int byteIndex)
         {
-            Debug.Assert(0 <= byteIndex && byteIndex <= Length * sizeof(char));
```

Comment:
This (and a bunch of other places) seems to be more likely to have been done in the style of range checks, than the normal assignment concerns.  Assuming we're changing it anyways.  Would be nice to have a better range check syntax, though.

Reply:
Yeah, it makes me slightly sad to lose this pattern. because I find it easier to quickly see it's correct. Another example
```
<             Debug.Assert(0 <= startPos && startPos <= s.Length);
>             Debug.Assert(startPos >= 0 && startPos <= s.Length);
```

Reply:
Another random one
```
-            Debug.Assert(0 <= charsConsumedSoFar && charsConsumedSoFar <= originalCharCount, "Caller should've checked this condition.");
+            Debug.Assert(charsConsumedSoFar >= 0 && charsConsumedSoFar <= originalCharCount, "Caller should've checked this condition.");
```

Reply:
Maybe we can get stylecop to change its detection to ignore range checks?

Someone flagged it here:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2830

Reply:
I doubt it - that would upset the other camp 😉 I think we can live with it, since we mix both already.